### ISSUE TITLE
import error

### ISSUE THREAD

**Issue by basz** [gcollazo/ember-cli-showdown#58]
after installing `ember serve` doesn't run anymore. Any idea what this can be?

```
while import vendor/showdown.js: found an unknown transformation name amd. Available transformNames are: 

Error: while import vendor/showdown.js: found an unknown transformation name amd. Available transformNames are: 
    at options.using.forEach.entry (/Users/bas/Documents/Projects/PLHW.ProjectHealthyFeet/applications/hf-app-lab/node_modules/ember-cli/lib/broccoli/ember-app.js:1688:19)
    at Array.forEach (native)
    at EmberApp._import (/Users/bas/Documents/Projects/PLHW.ProjectHealthyFeet/applications/hf-app-lab/node_modules/ember-cli/lib/broccoli/ember-app.js:1679:23)
    at EmberApp.import (/Users/bas/Documents/Projects/PLHW.ProjectHealthyFeet/applications/hf-app-lab/node_modules/ember-cli/lib/broccoli/ember-app.js:1652:10)
    at Class.import (/Users/bas/Documents/Projects/PLHW.ProjectHealthyFeet/applications/hf-app-lab/node_modules/ember-cli/lib/models/addon.js:677:15)
    at Class.included (/Users/bas/Documents/Projects/PLHW.ProjectHealthyFeet/applications/hf-app-lab/node_modules/ember-cli-showdown/index.js:15:16)
    at Class.superWrapper [as included] (/Users/bas/Documents/Projects/PLHW.ProjectHealthyFeet/applications/hf-app-lab/node_modules/ember-cli/node_modules/core-object/lib/assign-properties.js:34:20)
    at project.addons.project.addons.filter.addon (/Users/bas/Documents/Projects/PLHW.ProjectHealthyFeet/applications/hf-app-lab/node_modules/ember-cli/lib/broccoli/ember-app.js:545:17)
    at Array.filter (native)
    at EmberApp._notifyAddonIncluded (/Users/bas/Documents/Projects/PLHW.ProjectHealthyFeet/applications/hf-app-lab/node_modules/ember-cli/lib/broccoli/ember-app.js:542:47)
rattletrap:hf-app-lab bas$ 


```

**Comment by basz** [comment]
somewhere here.

https://github.com/gcollazo/ember-cli-showdown/blob/master/index.js#L15-L17

the transformations map empty, see https://github.com/ember-cli/ember-cli/blob/f46be84f38a0c790f864d5b982e675c206f59a45/lib/broccoli/ember-app.js#L1708-L1711

This feels like I am missing something here...

**Comment by jasonmit** [comment]
What version of ember-cli?

**Comment by basz** [comment]
Where your demo app? could not find it...

```
{
  "name": "plhw-client",
  "description": "Client hf-app-lab application to hf-app-api",
  "private": true,
  "directories": {
    "doc": "doc",
    "test": "tests"
  },
  "scripts": {
    "build": "ember build",
    "start": "ember server",
    "test": "ember test"
  },
  "repository": "",
  "engines": {
    "node": "^4.5 || 6.* || >= 7.*"
  },
  "author": "Bas Kamer <xxxxx>",
  "license": "LicenseRef-LICENSE.md",
  "devDependencies": {
    "bootstrap": "^4.0.0-beta",
    "broccoli-asset-rev": "^2.6.0",
    "broccoli-funnel": "^2.0.1",
    "broccoli-merge-trees": "^2.0.0",
    "ember-ajax": "^3.0.0",
    "ember-array-contains-helper": "^1.3.2",
    "ember-awesome-macros": "^0.40.0",
    "ember-best-language": "^0.0.3",
    "ember-body-class": "^0.4.0",
    "ember-bootstrap": "kaliber5/ember-bootstrap#bs4-beta",
    "ember-bootstrap-cp-validations": "^1.0.0-alpha.0",
    "ember-browserify": "^1.2.0",
    "ember-buffered-proxy": "^0.7.0",
    "ember-can": "^0.8.4",
    "ember-cli": "^2.16.0-beta.1",
    "ember-cli-app-version": "^3.1.0",
    "ember-cli-autoprefixer": "^0.8.0",
    "ember-cli-babel": "^6.8.2",
    "ember-cli-bootstrap-datepicker": "^0.5.6",
    "ember-cli-content-security-policy": "^1.0.0",
    "ember-cli-copyable": "^0.9.6",
    "ember-cli-dependency-checker": "^2.0.1",
    "ember-cli-deploy": "^1.0.1",
    "ember-cli-deploy-build": "^1.1.0",
    "ember-cli-deploy-display-revisions": "^1.0.0",
    "ember-cli-deploy-json-config": "^1.0.0",
    "ember-cli-deploy-revision-data": "^1.0.0",
    "ember-cli-deploy-rsync-assets": "^0.2.1",
    "ember-cli-deploy-slack": "^1.0.0",
    "ember-cli-deploy-ssh-index": "^0.2.3",
    "ember-cli-eslint": "^4.0.0",
    "ember-cli-fastboot": "^1.0.5",
    "ember-cli-gravatar": "^3.9.0",
    "ember-cli-htmlbars": "^2.0.3",
    "ember-cli-htmlbars-inline-precompile": "^1.0.2",
    "ember-cli-inject-live-reload": "^1.7.0",
    "ember-cli-moment-shim": "^3.5.0",
    "ember-cli-qunit": "^4.0.0",
    "ember-cli-release": "^1.0.0-beta.2",
    "ember-cli-sass": "^7.0.0",
    "ember-cli-shims": "^1.2.0-beta.2",
    "ember-cli-sri": "^2.1.1",
    "ember-cli-uuid": "^0.4.1",
    "ember-cookies": "^0.0.13",
    "ember-cp-validations": "^3.4.1",
    "ember-crumbly": "^1.0.7",
    "ember-data": "~2.15.0",
    "ember-data-model-fragments": "^2.14.0",
    "ember-data-url-templates": "^0.4.0",
    "ember-deferred-content": "^0.2.0",
    "ember-element-resize-detector": "^0.1.5",
    "ember-exif-shim": "^0.1.2",
    "ember-export-application-global": "^2.0.0",
    "ember-feature-flags": "^3.0.0",
    "ember-fetch": "^3.4.0",
    "ember-file-upload": "^2.1.1",
    "ember-flex-menu": "^0.0.16",
    "ember-font-awesome": "^4.0.0-alpha.4",
    "ember-g-map": "^0.0.25",
    "ember-i18n": "^5.0.2",
    "ember-i18n-cp-validations": "^3.0.2",
    "ember-light-table": "^1.10.0",
    "ember-line-clamp": "^0.1.3",
    "ember-link-action": "^0.0.36",
    "ember-load-initializers": "^1.0.0",
    "ember-local-storage": "^1.4.0",
    "ember-moment": "^7.4.1",
    "ember-multiselect-checkboxes": "^0.10.3",
    "ember-notify": "^5.2.1",
    "ember-progress-bar": "^0.0.7",
    "ember-qrcode": "^0.0.4",
    "ember-radio-button": "^1.1.1",
    "ember-resize": "^0.0.17",
    "ember-resolver": "^4.0.0",
    "ember-responsive": "^2.0.4",
    "ember-route-action-helper": "^2.0.6",
    "ember-scroll-to": "^0.6.4",
    "ember-shepherd": "^2.7.0",
    "ember-simple-auth": "^1.4.0",
    "ember-source": "~2.15.0",
    "ember-state-services": "^4.0.0",
    "ember-tether": "^1.0.0-beta.0",
    "ember-truth-helpers": "^1.3.0",
    "ember-watson": "^0.9.1",
    "emberx-select": "^3.1.0",
    "eslint-plugin-ember-suave": "^1.0.0",
    "liquid-fire": "^0.28.0",
    "loader.js": "^4.6.0",
    "node-zopfli": "^2.0.2",
    "outdated": "^0.1.1",
    "sockjs-client": "^1.1.4",
    "ui-navigator-service": "^0.3.11",
    "webstomp-client": "^1.0.7"
  },
  "dependencies": {
    "ember-cli-showdown": "^4.1.0"
  },
  "browserslist": {
    "production": [
      "last 2 version",
      "ie 10"
    ],
    "testing": [
      "last 2 versions"
    ],
    "development": [
      "last 2 versions"
    ]
  },
  "version": "0.22.2"
}
```

**Comment by jasonmit** [comment]
@basz unsure what you mean, the dummy app within the addon is the demo app.  But if you can reproduce this bug in a demo project and push it to github that would be helpful.  The dependencies you have listed out likely won't help me much.

**Comment by basz** [comment]
ok right... :-) Turns out I was using ember-cli v 2.16.0-beta1. Downgrading to 2.15.1 solves this issue

**Comment by basz** [comment]
Thanks you, i'll close this one for you.

If you're interested, this is what I'm using it for... https://gist.github.com/basz/49369483fcda33481eec24398e8a734f, localised markdown... 

One more question you might be able to answer; if i use links inside the markdown which point towards routes inside the application, the whole application obviously reloads. Would you know of a (recommended) to prevent such behaviour? I've looked at the link-to helper which adds a 'one' event listener to its rendered html and a stopPropagation. Did you ever deal with anchors inside an ember app coming from showdown?

**Comment by jasonmit** [comment]
> I've looked at the link-to helper which adds a 'one' event listener to its rendered html and a stopPropagation. 

That's how I would achieve this behavior.  This is also how ember-href-to is implemented which might be a decent reference https://github.com/intercom/ember-href-to/blob/master/app/instance-initializers/ember-href-to.js 

> Did you ever deal with anchors inside an ember app coming from showdown?

I haven't personally.

**Comment by basz** [comment]
great tip, thanks! that href-to addons works on any anchor that isn't an ember view in itself. I needed to change nothing. +1

**Comment by john-griffin** [comment]
Looks like this is still an issue in 2.16 stable.

**Comment by basz** [comment]
confirmed to be solved in 2.16.2
